astropy/units/tests/test_quantity_non_ufuncs.py

```diff
@@ -458,6 +465,34 @@ def test_arange_unit_from_stop(self):
         assert b.unit == u.min
         assert_array_equal(a.value, b.value)
 
+    def test_empty(self):
```

Comment:
You could factor out the test by making a new class
```
class TestArrayCreation(BasicTestSetup):
    def check(self, func, *args, **kwargs):
        o = func(*args, **kwargs, like=self.q)
        assert type(o) is type(self.q)
        assert o.unit == self.q.unit
        expected = func(*args, **kwargs) << self.q.unit
        assert_array_equal(o, expected)
```

Then only for `full` do you need to keep the test with `fill_value` with a unit.

Reply:
for reference, I've integrated this suggestion into #17125